frontend/packages/sdk/src/contracts/PinkContract.ts

```diff
@@ -82,32 +95,54 @@ class PinkContractSubmittableResult extends ContractSubmittableResult {
     this.#registry = registry
   }
 
-  async waitFinalized(timeout: number = 120_000) {
+  async waitFinalized(
+    predicate?: () => Promise<boolean>,
+    options?: { timeout?: number; blocks?: number }
+  ): Promise<void> {
```

Comment:
This update is base on the feedback from @kingsleydon . Now we can send contract transaction and wait for it finalize like this:
```ts
const { output: counts } = await contractPromise.query.totalBadge<u8>(cert.address, { cert })
const tx = await signAndSend(contractPromise.tx.newBadge(txOptions, 'New badge 01'), pair)
await tx.waitFinalized(async () => {
  return contractPromise.query.totalBadge<u8>(cert, address, { cert }) > counts
})
```